src/DockWidgetTab.cpp

```diff
@@ -628,14 +641,17 @@ QString CDockWidgetTab::text() const
 //============================================================================
 void CDockWidgetTab::mouseDoubleClickEvent(QMouseEvent *event)
 {
-	// If this is the last dock area in a dock container it does not make
-	// sense to move it to a new floating widget and leave this one
-	// empty
-	if ((!d->DockArea->dockContainer()->isFloating() || d->DockArea->dockWidgetsCount() > 1)
-		&& d->DockWidget->features().testFlag(CDockWidget::DockWidgetFloatable))
-	{
-        d->saveDragStartMousePosition(internal::globalPositionOf(event));
-		d->startFloating(DraggingInactive);
+	if (event->button() == Qt::LeftButton) {
```

Comment:
Double click would trigger while closing tabs very fast, I don't think it was intended to float with any button anyway.